private.tf

```diff
@@ -0,0 +1,57 @@
+module "private_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.3.0"
+  namespace  = "${var.namespace}"
+  name       = "${var.name}"
+  stage      = "${var.stage}"
+  delimiter  = "${var.delimiter}"
+  tags       = "${var.tags}"
+  attributes = ["${compact(concat(var.attributes, list("private")))}"]
+}
+
+resource "aws_subnet" "private" {
+  count             = "${var.type == "private" ? length(var.availability_zones) : 0}"
+  vpc_id            = "${var.vpc_id}"
+  availability_zone = "${element(var.availability_zones, count.index)}"
+  cidr_block        = "${cidrsubnet(var.cidr_block, ceil(log(var.max_subnets, 2)), count.index)}"
+
+  tags = {
+    "Name"      = "${module.private_label.id}${var.delimiter}${element(var.availability_zones, count.index)}"
+    "Stage"     = "${module.private_label.stage}"
+    "Namespace" = "${module.private_label.namespace}"
+    "AZ"        = "${element(var.availability_zones, count.index)}"
+    "Type"      = "${var.type}"
+  }
+}
+
+resource "aws_route_table" "private" {
+  count  = "${var.type == "private" ? length(var.availability_zones) : 0}"
+  vpc_id = "${var.vpc_id}"
+
+  tags = {
+    "Name"      = "${module.private_label.id}${var.delimiter}${element(var.availability_zones, count.index)}"
+    "Stage"     = "${module.private_label.stage}"
+    "Namespace" = "${module.private_label.namespace}"
+  }
+}
+
+resource "aws_route" "private" {
```

Comment:
This should be optional. If no `ngw_id` passed, do not create the route. E.g. if simple VPC with only RDS or elasticache, does not need a ngw. I have a current use case for this.